unit/unittype.go

```diff
@@ -130,6 +130,31 @@ var (
 // dimension is zero. Dimensions is used in conjunction with New.
 type Dimensions map[Dimension]int
 
+func (d Dimensions) clone() Dimensions {
+	if len(d) == 0 {
+		return nil
+	}
+	c := make(Dimensions, len(d))
+	for dim, pow := range d {
+		if pow != 0 {
+			c[dim] = pow
+		}
+	}
+	return c
+}
+
+func (d Dimensions) matches(o Dimensions) bool {
+	if len(d) != len(o) {
```

Comment:
There should be a comment somewhere (maybe in Unit.DimensionsMatch) that we require zero values to be deleted as keys. Just something so that if we change implementations in future there's a warning that we have some counter-intuitive behavior.

Reply:
Done